### DISCUSSION TITLE
Request dxdt in the output file

### DISCUSSION THREAD

**Discussion by Mutaz94** [metrumresearchgroup/mrgsolve#880] [Q&A] [answered]
Hi @kylebaron, 

Can we request the rate of change for each compartment (dxdt_comp) in the output file? 

Thanks!

**Comment by kylebaron** [comment, marked as answer]
Hi @Mutaz94  - 

You can copy it to a different name and then capture it


```c
[ des ] 

dxdt_depot = -ka*depot; 

double rate_depot = dxdt_depot;

[ capture] rate_depot

```

But just be careful using it; it's really for the solver to use and might not always be what you expect; no guarantees. 

Kyle

**Reply by Mutaz94** [reply]
Thanks for the prompt reply!! Thats awesome, its similar to NONMEM syntax. What do you recommend in case of calculating the rate of change (slope) at different times?

**Reply by kylebaron** [reply]
Hi @Mutaz94 - this isn't explicitly a function of time but rather amount in the compartments; so at any time the rate out of `depot` is `ka*depot`.  Also, keep in mind that there are possibly infusions that will get added to the `dxdt_cmt` items ... for example, if there is an infusion into the `depot` compartment, at the time you calculate this you'll need to know that too.

**Reply by Mutaz94** [reply]
Make sense. I think I can give it a try with different scenarios and see if that plausible. Thank you so much!
